rezerwacje-front/src/layouts/Main/Main.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+import clsx from 'clsx';
+import { makeStyles, useTheme } from '@material-ui/styles';
+import { useMediaQuery } from '@material-ui/core';
+
+import { Sidebar, Topbar } from './components';
+
+import { Switch, Route } from 'react-router-dom';
+import {
+  Home as HomeView,
+  Rooms as RoomsView
+} from '../../views';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    paddingTop: 56,
+    height: '100%',
+    [theme.breakpoints.up('sm')]: {
+      paddingTop: 64
+    }
+  },
+  shiftContent: {
+    paddingLeft: 240
+  },
+  content: {
+    height: '100%'
+  }
+}));
+
+const Routes = () => {
+  console.log("Main router called");
+  return (
+    <Switch>
+      <Route
+        exact
+        path="/home"
+        render={() => <HomeView/>} 
+      />
+      <Route 
+        exact
+        path="/rooms"
+        render={() => <RoomsView/>} 
+      />
+    </Switch>
+  );
+};
+
+const MainLayout = () => {
+
+  const classes = useStyles();
+  const theme = useTheme();
+  const isDesktop = useMediaQuery(theme.breakpoints.up('lg'), {
+    defaultMatches: true
+  });
+
+  const [openSidebar, setOpenSidebar] = useState(false);
+
+  const handleSidebarOpen = () => {
+    setOpenSidebar(true);
+  };
+
+  const handleSidebarClose = () => {
+    setOpenSidebar(false);
+  };
+
+  const shouldOpenSidebar = isDesktop ? true : openSidebar;
+
+  return (
+    <div
+      className={clsx({
+        [classes.root]: true,
+        [classes.shiftContent]: isDesktop
+      })}
+    >
+      <Topbar onSidebarOpen={handleSidebarOpen} />
+      <Sidebar
+        onClose={handleSidebarClose}
+        open={shouldOpenSidebar}
+        variant={isDesktop ? 'persistent' : 'temporary'}
+      />
+      <main className={classes.content}>
+        <Routes/>
```

Comment:
- [ ] Dependency inversion
Layout has nothing to do with routing. It should only display given children in a specific layout.